engine/class_modules/sc_shaman.cpp

```diff
@@ -4905,9 +4907,53 @@ struct static_discharge_t : public shaman_spell_t
 {
   static_discharge_t( shaman_t* player, const std::string& options_str )
     : shaman_spell_t( "static_discharge", player, player->talent.static_discharge, options_str )
+  {
+    affected_by_master_of_the_elements = false;
+  }
+
+  void execute() override
+  {
+    shaman_spell_t::execute();
+
+    p()->buff.static_discharge->trigger();
+  }
+
+  bool ready() override
+  {             //TODO: Implement Lightning Shield
+    if ( false) //!p()->buff.lightning_shield->up() )
     {
-      // placeholder
+      return false;
     }
+    else
+    {
+      return shaman_spell_t::ready();
+    }
+  }
+};
+
+struct static_discharge_tick_t : public shaman_spell_t
+{
+  static_discharge_tick_t( shaman_t* player)
+    : shaman_spell_t( "static discharge tick", player, player->find_spell( 342244 ) )
+  {
+    background = true;
+  }
+
+  void execute() override
+  { 
+    target_cache.is_valid          = false;
+    std::vector<player_t*> targets = target_list();
+    auto it                        = std::remove_if( targets.begin(), targets.end(), [ this ]( player_t* target ) {
+      return !this->td( target )->dot.flame_shock->is_ticking();
+    } );
+
+    targets.erase( it, targets.end() );
+    if ( targets.size() == 0 )
+      return;
```

Comment:
I think adding a print_debug statement for the case where no targets are available could help a lot here for any potential problems.

If this can happen in normal settings and not just edge cases you might even want to set up a proc_t for tracking how often static discharge got wasted because no target was available.